tcl/org.testeditor.tcl.dsl.tests/src/org/testeditor/tcl/dsl/tests/AbstractTclTest.xtend

```diff
@@ -13,13 +13,33 @@
 package org.testeditor.tcl.dsl.tests
 
 import com.google.inject.Module
+import com.google.inject.Provider
 import java.util.List
+import javax.inject.Inject
+import org.eclipse.emf.ecore.EObject
+import org.eclipse.xtext.resource.XtextResourceSet
+import org.junit.Before
+import org.testeditor.aml.AmlModel
 import org.testeditor.dsl.common.testing.AbstractTest
+import org.testeditor.tcl.TclModel
 import org.testeditor.tcl.dsl.TclRuntimeModule
 import org.testeditor.tcl.dsl.TclStandaloneSetup
+import org.testeditor.tsl.TslModel
+import org.testeditor.tcl.util.TclModelUtil
 
 abstract class AbstractTclTest extends AbstractTest {
 
+	@Inject protected Provider<XtextResourceSet> resourceSetProvider
+	@Inject protected XtextResourceSet resourceSet
+	
+	@Inject extension TclModelUtil
+
+	@Before
+	def void setUpResourceSet() {
+		resourceSet = resourceSetProvider.get
+		resourceSet.classpathURIContext = this		
```

Comment:
I don't think that this is a good place for this initialization. Instead, this could be extracted in a helper class that is add as an extension provider to the subclasses that need it.

A formatter test, for example, does not need a given resource set.